readme and methods adjustment

denarch22 opened the pull request:
## 🛠️ Changes
The README file is adjusted, the code is adjusted to remove the raster and an adjustment is made to validate the response when uploading the items since the error was not evident.

## 📝 Associated issues
[Ajuste en cargue y propiedades de las colecciones](https://github.com/PEM-Humboldt/stac-data-tools/pull/19)

denarch22 committed:
readme and methods adjustment

The README file is adjusted, the code is adjusted to remove the raster and an adjustment is made to validate the response when uploading the items since the error was not evident.

erikasv commented on README.md:
```diff
@@ -60,24 +60,77 @@ Para crear una colección siga los siguientes pasos:
+  
+- `-c`, `--collection` (opcional): Define el nombre de la colección. Si no se proporciona, se tomará el `id` del archivo `collection.json`.
+  
+- `-v`, `--validate-only` (opcional): Si se incluye, solo se validará la colección sin cargarla.
```
this option is not longer available. Now this is the command validate

erikasv commented on README.md:
```diff
@@ -60,24 +60,77 @@ Para crear una colección siga los siguientes pasos:
+
+## Eliminar una Colección
+
+Para eliminar una colección existente de Azure, ejecuta el siguiente comando:
```
this will remove the collection from azure as well as from the stac

erikasv commented on src/utils/stac_rest.py:
```diff
@@ -1,19 +1,26 @@
 
+    except requests.exceptions.RequestException as e:
+        logger.error(f"Error during post or put: {e}")
+        return None
```
I don't think it's a good idea to return None here. Why did you di this instead or raising the error?

denarch22 replied:
I left it for the moment to do the tests, I forgot to make the adjustment

erikasv commented on src/utils/storage.py:
```diff
@@ -28,6 +33,23 @@ def remove_file(self, file_path):
         """
         Remove a blob from Azure Blob Storage
         """
+        if file_path.startswith("https://"):
+            parsed_url = parse.urlparse(file_path)
+            file_path = parsed_url.path.lstrip("/")
+
+        container_name = self.container_client.container_name
+
+        if file_path.startswith(f"{container_name}/"):
+            file_path = file_path[len(f"{container_name}/") :]
+
```
I don't understand the purpose of these. Doesn't his function will always receive the full url (with https)

denarch22 replied:
Explanation of the Code
This code ensures that the correct internal path of a blob is used when attempting to delete it from Azure Blob Storage. Below is a detailed explanation of why each part of the logic is necessary:

Extracting the Path from the URL:

if file_path.startswith("https://"):

    parsed_url = parse.urlparse(file_path)
    file_path = parsed_url.path.lstrip("/")

Why:
If the function receives a full URL (e.g., https://staccatalog.blob.core.windows.net/cog-test/LossPersistence/2016-2021.tif), it needs to extract the path component to use it with Azure's get_blob_client().

Example: From the URL, it extracts the path:

cog-test/LossPersistence/2016-2021.tif.

lstrip("/") ensures any leading slashes are removed, leaving the path clean.

---------------
Removing the Container Name from the Path:

container_name = self.container_client.container_name

if file_path.startswith(f"{container_name}/"):
    file_path = file_path[len(f"{container_name}/") :]

Why:

Azure Blob Storage expects only the internal path of the blob (relative to the container) when accessing it. If the extracted path includes the container name, it must be removed.

Example:
If the path is cog-test/LossPersistence/2016-2021.tif and the container name is cog-test, the path is adjusted to:
LossPersistence/2016-2021.tif.

erikasv replied:
yeap, thanks, i was confused with the two ifs thinking in the second as an `elif` :sweat_smile: :see_no_evil:

cazapatamar commented on README.md:
```diff
@@ -60,24 +60,77 @@ Para crear una colección siga los siguientes pasos:
+
+### Parámetros:
+
+- `--collection` (obligatorio): Nombre de la colección a eliminar.
```
Please add `-c` option like above in order to let the user know it is possible to use both `-c` and `--collection`

cazapatamar commented on README.md:
```diff
@@ -60,24 +60,77 @@ Para crear una colección siga los siguientes pasos:
+
+---
+
+## Validar una Colección
```
Please add a subsection here to describe the parameters that the user can use for validation, to standardize with the other sections (load and remove a collection)

cazapatamar reviewed (changes requested):
I uploaded a full collection with 4 items for loss and persistence (name: CAZMTest-15102024).

- The collection info in the STAC is correct.
- The TIFFs are converted to COG, uploaded to Azure, and attached to items as assets.

The only issue I found is that the properties (colors, classes, values) were not correctly created for each item. The collection you previously uploaded, @denarch22, LossPersistence, does have these properties. Is it necessary to make any changes to the collection.json file when uploading the collection?

![imagen](https://github.com/user-attachments/assets/12463056-c035-40f9-a181-82a3b827a903)

In addition, the functions to validate and remove a collection are OK.

erikasv commented:
Also remove the '>' I'm the changes description (it's explicitly said in the PR template 

And change the associated issue link for the GitHub link (the goal of this link is to allow people outside zenhub to see how an issue was resolved)

denarch22 commented:
> Subí una colección completa con 4 elementos para pérdida y persistencia (nombre: CAZMTest-15102024).
> 
> * La información de la colección en el STAC es correcta.
> * Los TIFF se convierten en COG, se cargan en Azure y se adjuntan a los elementos como recursos.
> 
> El único problema que encontré es que las propiedades (colores, clases, valores) no se crearon correctamente para cada elemento. La colección que cargó anteriormente, , LossPersistence, tiene estas propiedades. ¿Es necesario realizar algún cambio en el archivo collection.json al cargar la colección?
> 
> ![imagen](https://private-user-images.githubusercontent.com/19979979/376872232-12463056-c035-40f9-a181-82a3b827a903.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxNzU1MjksIm5iZiI6MTcyOTE3NTIyOSwicGF0aCI6Ii8xOTk3OTk3OS8zNzY4NzIyMzItMTI0NjMwNTYtYzAzNS00MGY5LWExODEtODJhM2I4MjdhOTAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE3VDE0MjcwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2NjFlMjE1ZWY4MzkzMWJjZTllYjJjZmYzYTYwMmEzNmU3ZGU3OTM3Mjc4ZWQ5MmRiM2Q4MTAzMTlmM2MxOTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LydbIPXbcUZHj4DS1cRher2X2BaNRSq0_SJlDXkgYww)
> 
> Además, las funciones para validar y eliminar una colección son correctas.

yes camilo apart from the one in collection which is mandatory, it is necessary to add it to the items, like this:
![image](https://github.com/user-attachments/assets/b8dfe401-46ee-4459-a7dd-41666bd0a8d7)

denarch22 committed:
readme and logs adjustment

readme and logs adjustment

denarch22 committed:
adjust format

adjust format

denarch22 committed:
adjust parameter

adjust parameter

erikasv commented:
> si camilo aparte del que esta en collection que es obligatorio, es necesario agregarlo en los items, asi: ![image](https://private-user-images.githubusercontent.com/172850334/377530041-b8dfe401-46ee-4459-a7dd-41666bd0a8d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxODAyMjYsIm5iZiI6MTcyOTE3OTkyNiwicGF0aCI6Ii8xNzI4NTAzMzQvMzc3NTMwMDQxLWI4ZGZlNDAxLTQ2ZWUtNDQ1OS1hN2RkLTQxNjY2YmQwYThkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxN1QxNTQ1MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNzJiODg0NDQ0MGUzYWE1MjNiZWFjNjU0ZDliZTIyZGY3N2JhZGUwMjU2ZjBjMDJkYWQ4NzgzZGI2OWYyZjczJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.O_c_JjNeqdLSbsF_SDFjdzFSsiw-t8BD7NwKVHT6w20)

@denarch22 the purpose of the collection spec is to simplify things for other researchers, so we need to keep the properties at the collection level, but internally those properties will be associated to each item, if each item has any other property it will be added to that specific item (or replace the one inherited from the collection). Also none of them are required in the spec.

Please add these clarifications in the spec.md file (that the properties in the collection level metadata will be associated to each item in the stac collection and that if the same property is set in the collection and in the item, the item will overwrite the collection one ). Keep this file in spanish

cazapatamar commented:
> > Subí una colección completa con 4 elementos para pérdida y persistencia (nombre: CAZMTest-15102024).
> > 
> > * La información de la colección en el STAC es correcta.
> > * Los TIFF se convierten en COG, se cargan en Azure y se adjuntan a los elementos como recursos.
> > 
> > El único problema que encontré es que las propiedades (colores, clases, valores) no se crearon correctamente para cada elemento. La colección que cargó anteriormente, , LossPersistence, tiene estas propiedades. ¿Es necesario realizar algún cambio en el archivo collection.json al cargar la colección?
> > ![imagen](https://private-user-images.githubusercontent.com/19979979/376872232-12463056-c035-40f9-a181-82a3b827a903.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxNzU1MjksIm5iZiI6MTcyOTE3NTIyOSwicGF0aCI6Ii8xOTk3OTk3OS8zNzY4NzIyMzItMTI0NjMwNTYtYzAzNS00MGY5LWExODEtODJhM2I4MjdhOTAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE3VDE0MjcwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2NjFlMjE1ZWY4MzkzMWJjZTllYjJjZmYzYTYwMmEzNmU3ZGU3OTM3Mjc4ZWQ5MmRiM2Q4MTAzMTlmM2MxOTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LydbIPXbcUZHj4DS1cRher2X2BaNRSq0_SJlDXkgYww)
> > Además, las funciones para validar y eliminar una colección son correctas.
> 
> si camilo aparte del que esta en collection que es obligatorio, es necesario agregarlo en los items, asi: ![image](https://private-user-images.githubusercontent.com/172850334/377530041-b8dfe401-46ee-4459-a7dd-41666bd0a8d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxODEyMjEsIm5iZiI6MTcyOTE4MDkyMSwicGF0aCI6Ii8xNzI4NTAzMzQvMzc3NTMwMDQxLWI4ZGZlNDAxLTQ2ZWUtNDQ1OS1hN2RkLTQxNjY2YmQwYThkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxN1QxNjAyMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYjA4MWE5N2U2NWViMDBiNTFhNWNkMTdlY2IzNGJiMjM3MTNkNjYxOTU2YjdjNzk4NjM2NDYxMDlmNjdjMzhhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.TrBupFqFp5S2Ne-P0-_4EyUdcojiYmAajaM4bMb2bQg)

Please update the section `Preparación de insumos` (if necessary) and associated files `collection.md` and `collection.example.json` with the new structure

denarch22 committed:
adjust spec

adjust spec

denarch22 committed:
validation adjustment collection

validation adjustment collection

cazapatamar commented on README.md:
````diff
@@ -60,24 +60,101 @@ Para crear una colección siga los siguientes pasos:
+python src/main.py create -f folder_name [-c collection_name] [-o]
+```
+
+# Parámetros:
````
Use ### instead of only one # in this title

cazapatamar commented on README.md:
````diff
@@ -60,24 +60,101 @@ Para crear una colección siga los siguientes pasos:
+python src/main.py validate -f folder_name [-c collection_name]
+```
+
+# Parámetros:
````
Use ### instead of only one # in this title

cazapatamar reviewed (changes requested):
Also, please add `metadata.properties.colors` to the specification of the collection in `collection.md`

denarch22 committed:
setting properties collection and documentation

setting properties collection and documentation

denarch22 committed:
adjust properties

Make the adjustment again in the collection properties since they were remaining at the same level as the other properties.

denarch22 commented:
According to the last meeting, I made the adjustment so that the properties would remain within the collection with the extra fields option.

![image](https://github.com/user-attachments/assets/52ee5b99-9c09-4823-9fd0-c8aa2b079dfe)

cazapatamar commented:
> Also, please add `metadata.properties.colors` to the specification of the collection in `collection.md`

Update: According to our last meeting, we realized it is not necessary to change `collection.md`

ManuelStardust reviewed (approved):
After testing everything works fine.
